### PR TITLE
sbt 1.0.0-M1 publishing related changes

### PR DESCRIPTION
**Pull request by eed3si9n**
Some changes I encountered while publishing 1.0.0-M1

/review @dwijnand, @Duhemm

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -1,11 +1,16 @@
 Migration notes
 ===============
 
-- `project/Build.scala` style based on `sbt.Build` is removed. Migrate to `build.sbt`.
-- `Project(...)` constructor is limited to just two parameters.
+- Build definition is based on Scala 2.11.8
+- Build.scala style builds are gone. Use multi-project `build.sbt`.
+- `Project(...)` constructor is restricted down to two parameters. Use `project` instead.
+- `sbt.Plugin` is also gone. Use auto plugins.
+- The incremental compiler, called Zinc, is uses class-based name hashing.
```
Comment (dwijnand) [inline MIGRATION.md]: uses or is using

File: notes/1.0.0-M1.markdown [hunk 2]
```diff
@@ -0,0 +1,23 @@
+This is the first milestone (beta) release of sbt 1.0.
+There's no binrary compatibility with sbt 0.13 or other future versions.
```
Comment (dwijnand) [inline notes/1.0.0-M1.markdown]: binary

**Comment by dwijnand** [comment]
Why is the launchconfig moved from sbt to xsbt?

**Comment by dwijnand** [comment]
LGTM

**Comment by eed3si9n** [comment]
> Why is the launchconfig moved from sbt to xsbt?

One of the reasons people would want to use conscript on sbt/sbt is for `scalas` (script runner) not for `sbt`. Renaming to `xsbt` prevent conscript from hiding `sbt` the user might have installed via our installer or sbt-extras.

**Pull request merged**